examples/energy-management-app/energy-management-common/energy-evse/src/EnergyEvseDelegateImpl.cpp

```diff
@@ -1661,26 +1662,39 @@ void EvseSession::StartSession(int64_t chargingMeterValue, int64_t dischargingMe
     mSessionEnergyCharged    = MakeNullable(static_cast<int64_t>(0));
     mSessionEnergyDischarged = MakeNullable(static_cast<int64_t>(0));
 
-    MatterReportingAttributeChangeCallback(mEndpointId, EnergyEvse::Id, SessionID::Id);
-    MatterReportingAttributeChangeCallback(mEndpointId, EnergyEvse::Id, SessionDuration::Id);
-    MatterReportingAttributeChangeCallback(mEndpointId, EnergyEvse::Id, SessionEnergyCharged::Id);
-    MatterReportingAttributeChangeCallback(mEndpointId, EnergyEvse::Id, SessionEnergyDischarged::Id);
+    MatterReportingAttributeChangeCallback(aEndpointId, EnergyEvse::Id, SessionID::Id);
+    MatterReportingAttributeChangeCallback(aEndpointId, EnergyEvse::Id, SessionDuration::Id);
+    MatterReportingAttributeChangeCallback(aEndpointId, EnergyEvse::Id, SessionEnergyCharged::Id);
+    MatterReportingAttributeChangeCallback(aEndpointId, EnergyEvse::Id, SessionEnergyDischarged::Id);
 
     // Write values to persistent storage.
-    ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, EnergyEvse::Id, SessionID::Id);
+    ConcreteAttributePath path = ConcreteAttributePath(aEndpointId, EnergyEvse::Id, SessionID::Id);
     GetSafeAttributePersistenceProvider()->WriteScalarValue(path, mSessionID);
 
     // TODO persist mStartTime
     // TODO persist mSessionEnergyChargedAtStart
     // TODO persist mSessionEnergyDischargedAtStart
 }
 
+/**
+ * @brief This function updates the session information at the unplugged event
+ *
+ * @param chargingMeterValue    - The current value of the energy meter (charging) in mWh
```

Comment:
if you are adding doccoments, you should add the endpoint id.

At the same time however, I would prefer doc comments in headers only and not in cpp files (when looking at an interface, I am more likely to open the header not the implementation details)

Reply:
Ok changed

Reply:
However I've left the doc strings in the cpp for now since I think this is a preference - unless it's written in some coding guidelines I've missed? I prefer to see headers in the .cpp as well.